src/hooks/useSnackbar/useSnackbar.tsx

```diff
@@ -1,76 +1,98 @@
-import { snackbarTransitions } from '@/shared/components/Snackbar'
+import React, { createContext, useContext, useState, useRef, useEffect } from 'react'
+import { CSSTransition, TransitionGroup } from 'react-transition-group'
+import styled from '@emotion/styled'
+import faker from 'faker'
 import Snackbar from '@/shared/components/Snackbar/Snackbar'
-import { transitions } from '@/shared/theme'
-import { Global } from '@emotion/react'
-import React, { createContext, useContext, useEffect, useState } from 'react'
-import { CSSTransition } from 'react-transition-group'
+import { transitions, sizes } from '@/shared/theme'
 
 export type DisplaySnackbarArgs = {
   time?: number
-  variant?: 'success' | 'error' | 'info'
-  buttonText?: string
+  variant?: 'primary' | 'secondary'
+  icon?: 'success' | 'error' | 'info'
   message: string
+  subMessage?: string
+  actionText?: string
+  onActionClick?: () => void
 }
 
+type SnackbarsState = {
+  id: string
+} & Omit<DisplaySnackbarArgs, 'time'>
+
 type SnackbarContextValue = {
   displaySnackbar: (args: DisplaySnackbarArgs) => void
-  closeSnackbar: () => void
+  closeSnackbar: (id: string) => void
 }
 
 const SnackbarContext = createContext<SnackbarContextValue | undefined>(undefined)
 SnackbarContext.displayName = 'SnackbarContext'
 
+const SNACKBARS_LIMIT = 3
+
 export const SnackbarProvider: React.FC = ({ children }) => {
-  const [isVisible, setIsVisible] = useState(false)
-  const [snackbarOpts, setSnackbarOpts] = useState<DisplaySnackbarArgs | null>(null)
+  const [snackbars, setSnackbars] = useState<SnackbarsState[]>([])
+
+  // Snackbars state reference is needed for setTimeout to read
+  const snackbarsRef = useRef(snackbars)
+  snackbarsRef.current = snackbars
+
+  const displaySnackbar = ({ time, ...args }: DisplaySnackbarArgs) => {
+    const id = faker.random.uuid()
```

Comment:
`faker` has a _huge_ bundle size, we shouldn't use it in our runtime. Let's use some alternate method to generate an ID, like number of milliseconds or something. Matter of fact, it'd be best if we put that in `utils` since we already generate IDs in couple places so we can reuse that logic. `hooks/useDrafts/utils` already uses dome ID generation, we can use that as a start